### PR TITLE
OSX app notarization

### PR DESCRIPTION
**Pull request by wpotrzebowski**
conda_qt5_osx.yml is updated with libgfortran and clang_osx-64. No compiler is packaged though  at this stage, only some header files required for notarization.  Attempts to package compiler failed and require more work (to be addressed after 5.0.2). 

The notarzied app can be tested by downloading dmg file from SasView 5.0.2 draft (https://github.com/SasView/sasview/releases). 

Notrarization process cannot be fully automatized. Deployment process is described here: https://github.com/SasView/sasview/wiki/DevNotes_Processeses_Notarization.

### CODE REVIEW

**Comment by wpotrzebowski** [comment]
5.0 ready testing on OSX

**Comment by wpotrzebowski** [comment]
5.0 ready testing on OSX

**Comment by wpotrzebowski** [comment]
5.0 ready for testing on OSX

**Comment by wpotrzebowski** [comment]
5.0 ready for testing on OSX

File: build_tools/code_sign_osx.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""
+Listing all so and dylib files and signs them separetly.
+Required for notarization.
```
[the remark this answers is not in the document]
Comment (wpotrzebowski) [reply build_tools/code_sign_osx.py:3]: Good point but I couldn't bypass it. Each and every object inside app has to be signed for notarization to work.

File: build_tools/code_sign_osx.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+"""
+import glob
+import os
+
+so_list = glob.glob("SasView5.0.app/Contents/MacOS/**/*.so", recursive=True)
+dylib_list = glob.glob("SasView5.0.app/Contents/MacOS/**/*.dylib", recursive=True)
```
[the remark this answers is not in the document]
Comment (wpotrzebowski) [reply build_tools/code_sign_osx.py:9]: Indeed, I need to address naming. I also prefer plain SasView but what if someone wants to keep different versions in the same app folder? Then full tag would be preferred I suppose.

[the remark this answers is not in the document]
Comment (wpotrzebowski) [reply build_tools/code_sign_osx.py:9]: Also we need to make it consistent with other platforms

[the remark this answers is not in the document]
Comment (pkienzle) [reply build_tools/code_sign_osx.py:9]: When you drag the app from the dmg into a folder that already has a version then finder will ask you nicely if you want to "Keep Both", "Stop" or "Replace".  The installation process is different on the different platforms, so using a different app name should not be a problem.

[the remark this answers is not in the document]
Comment (wpotrzebowski) [reply build_tools/code_sign_osx.py]: I've changed app name to SasView5. I thought it is still appropriate when we release both 4 and 5

**Pull request merged**